src/libraries/Common/src/Interop/Interop.Ldap.cs

```diff
@@ -125,6 +126,12 @@ internal sealed class berval
         public IntPtr bv_val = IntPtr.Zero;
 
         public berval() { }
+
+        public berval(string value)
+        {
+            bv_len = value.Length;
+            bv_val = Marshal.StringToHGlobalAuto(value);
```

Comment:
Where is this memory going to be freed?

Reply:
This alloc should be in the caller, where it can be matched with a FreeHGlobal. This helper is not necessary , I think.

Reply:
Ah, OK. this was my first real foray into C# interop (and C). I thought I had read that the memory would be freed when the surrounding class was disposed, but I must have been half asleep when I was reading the documentation.